### ISSUE TITLE
aws-iam-no-policy-wildcards (AWS099) not ignored

### ISSUE THREAD

**Issue by zappallot** [aquasecurity/tfsec#952]
**Describe the bug**
The policy `aws-iam-no-policy-wildcards` or `AWS099` still comes up even when ignored.

**To Reproduce**
1. Run `tfsec .` and watch `aws-iam-no-policy-wildcards` come up as a result of the scan
2. Add `# tfsec:ignore:aws-iam-no-policy-wildcards` or `# tfsec:ignore:AWS099` as a comment to the resource.
3. Run `tfsec .` again.
4. Policy will still be printed as an error

**Expected behavior**
Policy is ignored for this resource.

**Screenshots**
```
> tfsec .

  Result 1

  [aws-iam-no-policy-wildcards][HIGH] Resource 'data.aws_iam_policy_document' defines a policy with wildcarded resources.
  <path>/iam.tf:120


     117 |     ]
     118 |
     119 | #tfsec:ignore:aws-iam-no-policy-wildcards
     120 |     resources = ["*"]    list: ["*"]
     121 |
     122 |     condition {
     123 |       test     = "StringEquals"

  Legacy ID:  AWS099
  Impact:     Overly permissive policies may grant access to sensitive resources
  Resolution: Specify the exact permissions required, and to which resources they should apply instead of using wildcards.

  More Info:
  - https://tfsec.dev/docs/aws/iam/no-policy-wildcards#aws/iam
  - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document
  - https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html
```

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04.2 LTS
 - TFSEC: v0.51.4

**Additional context**
The TF configuration in questions is a `data` item:
`data "aws_iam_policy_document" "something" {}`

I tried ignoring it with the comment above, below and inline. Both with the new and legacy ignore string.

**Comment by owenrumney** [comment]
Hey @zappallot  - a workaround is to add the ignore to the the resource that is using the policy document - but we're fixing that now

**Comment by zappallot** [comment]
Hi @owenrumney, thanks for the quick reply.
in my case, I pass the policy document to a module variable. 
I tried adding the ignore statement to the module like with other modules, but the policy is still reported.
Good to know it will be fixed. :-)

**Comment by liamg** [comment]
Hi @zappallot , this should be fixed for you in v0.52.0, please shout if not.
